widgets/src/input/text_select_region.rs

```diff
@@ -0,0 +1,64 @@
+use ribir_core::prelude::*;
+
+use super::handle::TextSelectEvent;
+
+/// A Widget that extends [`Text`] to emit text select event
+#[derive(Declare)]
+pub struct TextSelectRegion {
```

Comment:
This widget could be more versatile if it's simply called `SelectRegion` and not just for text. It should keep track of the selected area so that other parts can see what's selected. For text, it should calculate the glyph range. If the range covers a whole glyph, it's a selection; otherwise, it's a cursor position.